_posts/2022-11-07-road-to-quarkus-3.adoc

```diff
@@ -0,0 +1,76 @@
+  - *java.util.concurrent.Flow* - Time to evolve; Flow has been in the JDK since Java 9. Thus we want to move from Reactive Streams to java.util.concurrent.Flow.
+  - *Documentation* - Work started on better and more structured documentation following the https://diataxis.fr/[Diataxis] framework principles.
+  - *Tooling for updates* - going to be a new major version, we want to make the migration as smooth as possible. Thus we will provide migration tools.
+  - A new *gRPC server*, more flexible and easier to maintain
+  - A *revamped dev UI*
+  - <Your extension awesome feature/improvements here>
+
+## Release Plan
+
+We will continue the monthly cadence release of the 2.x version of Quarkus. Quarkus 3.x preview (alphas, betas) will be released in parallel. This blog marks the start of the first Quarkus 3 Alpha.
+
+It will take more than a month to integrate the change set we want and collect feedback (from users, and extension maintainers...) from the previous release. Thus, 3.x release cadence will be a bit slower than usual.
+
+Thus the intent is to have a time period of some months to allow continuous integration and start releasing a Quarkus Platform with the members that are ready to do and then continuously include more.
+
+Proposed plan:
+
+  - Start doing 3.0.0.Alpha core and platform releases at regular intervals to allow easy integration builds, starting first from 2.13.3 in the coming week
+  - 2.x releases will continue to happen on a monthly basis
+  - Announce Quarkus 3 on Quarkus blog when 3.0.0.Alphas start being useful for users to try (This Blog!)
+  - Feedback from and Outreach to extension contributors and users on readiness and challenges to help and adjust
+  - If all goes well, aim for a 3.0.0.Final in February 2023
+
+== Breakage? Keep calm!
+
+While the major version bump signals possible breaking changes, we still ask that all contributors still aim for a smooth upgrade experience. Changes like the move to Jakarta with its package renames have a big impact. We intend on doing what we can to make it so current extensions and user applications can migrate from Quarkus 2 to Quarkus 3 by providing tooling to perform the migration.
+
+Especially for the core API’s we will aim to allow such migration to be trivial and thus can be automated in a similar manner as how we’ve done automatic conversion of Quarkus 2.x to use Jakarta named packages.
+
+== Trying it out
+
+For now the easiest way to get started is using https://code.quarkus.io/?S=io.quarkus.platform%3A3.0[code.quarkus.io] or use the Quarkus cli:
+
+```
+quarkus app create --stream=3.0
+```
+
+Mind you that at this stage only parts of the Quarkus platform is present. Not all extensions have yet migrated to Jakarta packages.
+
+-- TBD: mention the extensions already committed or planning to be ready.
+
+For existing applications where all extensions are available try migrate from using `javax.` to `jakarta.` manually and report back if it works or not for you.
```

Comment:
Maybe add:
```
You can automate the transformation with the Eclipse Jakarta Transformer.
```
I think people should probably use this until we push our own tool because not all `javax` packages have changed. So if you do a search/replace for everything javax it will fail.